src/common/plugin/util/maximumFeeRateCalculator.ts

```diff
@@ -0,0 +1,19 @@
+import { EdgeCurrencyInfo } from 'edge-core-js/types'
+
+/**
+ * Calculates a maximum fee rate of 1 USD per vByte given a USD exchange rate
+ * (usdPrice).
+ */
+export const maximumFeeRateCalculator = (
+  currencyInfo: EdgeCurrencyInfo,
+  usdPrice: number
+): string | undefined => {
+  const { currencyCode, denominations } = currencyInfo
+  const primaryDenomination = denominations.find(
+    denom => denom.name === currencyCode
+  )
+  if (primaryDenomination == null) return undefined
+
+  const decimals: number = primaryDenomination.multiplier.length - 1
+  return Math.round((1 / usdPrice) * 10 ** decimals).toString()
```

Comment:
Just do the math with biggystring so you don't convert to and from strings/numbers.

This also ensures you never get scientific notation

```
import { div, mul, round } from 'biggystring

return round(mul(div('1', String(usdPrice), 16), primaryDenomination.multiplier))
```